pkg/blueprint/partitioning_customizations.go

```diff
@@ -0,0 +1,545 @@
+	MinSize uint64 `json:"minsize" toml:"minsize"`
+
+	BtrfsVolumeCustomization
+
+	VGCustomization
+
+	FilesystemTypedCustomization
+}
+
+// A filesystem on a plain partition or LVM logical volume.
+// Note the differences from [FilesystemCustomization]:
+//   - Adds a label.
+//   - Adds a filesystem type (fs_type).
+//   - Does not define a size. The size is defined by its container: a
+//     partition ([PartitionCustomization]) or LVM logical volume
+//     ([LVCustomization]).
+type FilesystemTypedCustomization struct {
+	Mountpoint string `json:"mountpoint" toml:"mountpoint"`
+	Label      string `json:"label,omitempty" toml:"label,omitempty"`
+	FSType     string `json:"fs_type,omitempty" toml:"fs_type,omitempty"`
+}
+
+// An LVM volume group with one or more logical volumes.
+type VGCustomization struct {
+	// Volume group name (optional, default will be automatically generated).
+	Name           string            `json:"name" toml:"name"`
+	LogicalVolumes []LVCustomization `json:"logical_volumes,omitempty" toml:"logical_volumes,omitempty"`
+}
+
+type LVCustomization struct {
+	// Logical volume name
+	Name string `json:"name,omitempty" toml:"name,omitempty"`
+
+	// Minimum size of the logical volume
+	MinSize uint64 `json:"minsize,omitempty" toml:"minsize,omitempty"`
+
+	FilesystemTypedCustomization
+}
+
+// Custom JSON unmarshaller for LVCustomization for handling the conversion of
+// data sizes (minsize) expressed as strings to uint64.
+func (lv *LVCustomization) UnmarshalJSON(data []byte) error {
+	var lvAnySize struct {
+		Name    string `json:"name,omitempty" toml:"name,omitempty"`
+		MinSize any    `json:"minsize,omitempty" toml:"minsize,omitempty"`
+		FilesystemTypedCustomization
+	}
+	if err := json.Unmarshal(data, &lvAnySize); err != nil {
+		return err
+	}
+
+	lv.Name = lvAnySize.Name
+	lv.FilesystemTypedCustomization = lvAnySize.FilesystemTypedCustomization
+
+	if lvAnySize.MinSize != nil {
+		size, err := decodeSize(lvAnySize.MinSize)
+		if err != nil {
+			return err
+		}
+		lv.MinSize = size
+	}
+	return nil
+}
+
+// A btrfs volume consisting of one or more subvolumes.
+type BtrfsVolumeCustomization struct {
+	Subvolumes []BtrfsSubvolumeCustomization
+}
+
+type BtrfsSubvolumeCustomization struct {
+	// The name of the subvolume, which defines the location (path) on the
+	// root volume (required).
+	// See https://btrfs.readthedocs.io/en/latest/Subvolumes.html
+	Name string `json:"name" toml:"name"`
+
+	// Mountpoint for the subvolume.
+	Mountpoint string `json:"mountpoint" toml:"mountpoint"`
+}
+
+// Custom JSON unmarshaller that first reads the value of the "type" field and
+// then deserialises the whole object into a struct that only contains the
+// fields valid for that partition type. This ensures that no fields are set
+// for the substructure of a different type than the one defined in the "type"
+// fields.
+func (v *PartitionCustomization) UnmarshalJSON(data []byte) error {
+	errPrefix := "JSON unmarshal:"
```

Comment:
What's the purpose of errPrefix if it is the same for all of the errors? I think it makes the code harder to read when jumping into the middle of it, and it also makes it impossible to grep for things like 'JSON unmarshal: foo failure at position bar'